pkg/controller/association/ca.go

```diff
@@ -24,40 +23,40 @@ type CASecret struct {
 	CACertProvided bool
 }
 
-// ElasticsearchCACertSecretName returns the name of the secret holding the certificate chain used
-// by the associated resource to establish and validate a secured HTTP connection to Elasticsearch.
-func ElasticsearchCACertSecretName(associated commonv1.Associated, associationName string) string {
+// ServiceCACertSecretName returns the name of the secret holding the certificate chain used
+// by the associated resource to establish and validate a secured HTTP connection to the target service.
+func ServiceCACertSecretName(associated commonv1.Associated, associationName string) string {
 	return associated.GetName() + "-" + associationName + "-ca"
 }
 
-// ReconcileCASecret keeps in sync a copy of the Elasticsearch CA.
-// It is the responsibility of the controller to set a watch on the ES CA.
-func (r *Reconciler) ReconcileCASecret(associated commonv1.Associated, es types.NamespacedName) (CASecret, error) {
-	publicESHTTPCertificatesNSN := certificates.PublicCertsSecretRef(esv1.ESNamer, es)
+// ReconcileCASecret keeps in sync a copy of the target service CA.
+// It is the responsibility of the association controller to set a watch on this CA.
+func (r *Reconciler) ReconcileCASecret(association commonv1.Association, namer name.Namer, service types.NamespacedName) (CASecret, error) {
+	servicePublicHTTPCertificatesNSN := certificates.PublicCertsSecretRef(namer, service)
 
 	// retrieve the HTTP certificates from ES namespace
-	var publicESHTTPCertificatesSecret corev1.Secret
-	if err := r.Get(publicESHTTPCertificatesNSN, &publicESHTTPCertificatesSecret); err != nil {
+	var servicePublicHTTPCertificatesSecret corev1.Secret
+	if err := r.Get(servicePublicHTTPCertificatesNSN, &servicePublicHTTPCertificatesSecret); err != nil {
 		if errors.IsNotFound(err) {
 			return CASecret{}, nil // probably not created yet, we'll be notified to reconcile later
 		}
 		return CASecret{}, err
 	}
 
-	labels := r.AssociationLabels(k8s.ExtractNamespacedName(associated))
+	labels := r.AssociationLabels(k8s.ExtractNamespacedName(association))
 	// Add the Elasticsearch name, this is only intended to help the user to filter on these resources
-	labels[eslabel.ClusterNameLabelName] = es.Name
+	labels[eslabel.ClusterNameLabelName] = service.Name
```

Comment:
would this set the ES cluster label to the Kibana TLS cert secret copy?

Reply:
Yes, good catch. I'm not sure it makes sense to add this label since it is already set on the original Secret. I would be in favor of removing it, wdyt ?

Reply:
I'm not sure myself actually. If we want to keep this label then it needs to be provided by a new field in the `AssociationInfo`. Not sure it is a bid deal, I can also do that.

Reply:
The label still feels useful to me? I think I'd be ok with adding a new field for it explicitly.